mars/services/task/analyzer/analyzer.py

```diff
@@ -31,6 +37,18 @@
 logger = logging.getLogger(__name__)
 
 
+def need_reassign_worker(op: OperandType) -> bool:
+    # NOTE(qinxuye): special process for reducer
+    # We'd better set reducer op's stage to reduce, however,
+    # in many case, we copy a reducer op from tileable op,
+    # then set stage as reducer one,
+    # it would be quite nasty to take over the __setattr__ and
+    # make reassign_worker True etc.
+    return op.reassign_worker or (
+        isinstance(op, MapReduceOperand) and op.stage == OperandStage.reduce
+    )
```

Comment:
Can we rename `reassign_worker` as `_reassign_worker` and create a property with default value given different operand classes instead?

Reply:
`reassign_worker` is recorded on `SchedulingHint` instead of operand itself, we cannot implement such a getter in this way.